llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -18774,7 +18774,7 @@ static SDValue performVecReduceAddCombineWithUADDLP(SDNode *N,
 static SDValue
 performActiveLaneMaskCombine(SDNode *N, TargetLowering::DAGCombinerInfo &DCI,
                              const AArch64Subtarget *ST) {
-  if (DCI.isBeforeLegalize())
+  if (DCI.isBeforeLegalize() && !!DCI.isBeforeLegalizeOps())
```

Comment:
nit: The `!!` looks a little odd. Is it possible to just use `DCI.isBeforeLegalizeOps()`?

Reply:
This was a mistake, it should be `!DCI.isBeforeLegalizeOps()`